### ISSUE TITLE
Bug: NoInjectDelegate false positive in Activity/Application

### ISSUE THREAD

**Issue by kirich1409** [androidbroadcast/Koin-Detekt#16]
## Summary

The `NoInjectDelegate` rule suggests using constructor injection instead of `by inject()`, but in Android `Activity` and `Application` classes **constructor injection is not possible** — the Android framework instantiates these classes via reflection using a no-arg constructor.

## Current Behavior

```kotlin
class MainActivity : AppCompatActivity(), KoinComponent {
    private val viewModel: MainViewModel by inject() // ← flagged
    private val analytics: Analytics by inject()      // ← flagged
}
```

**Detekt output:**
```
NoInjectDelegate - Lazy service locator, harder to test—Use constructor injection
```

## Expected Behavior

The rule should **not** flag `by inject()` usage in classes where constructor injection is fundamentally impossible:
- `Activity` / `AppCompatActivity` subclasses
- `Application` subclasses
- `Fragment` subclasses (though Fragment supports factory injection in some cases)
- `Service` / `BroadcastReceiver` subclasses

The rule should check the class hierarchy and skip reporting for known Android framework classes.

**Note:** The `NoKoinComponentInterface` rule already has an `allowedSuperTypes` config for this exact pattern. `NoInjectDelegate` could use a similar approach.

## Proposed Fix

Add an `allowedSuperTypes` configuration option (similar to `NoKoinComponentInterface`) that defaults to common Android classes:

```yaml
NoInjectDelegate:
  active: true
  allowedSuperTypes:
    - 'Application'
    - 'Activity'
    - 'AppCompatActivity'
    - 'Fragment'
    - 'Service'
    - 'BroadcastReceiver'
    - 'ViewModel'
```

## Impact

5 false positives in a real project using standard Android + Koin patterns.

## Environment

- Koin-Detekt 0.4.0
- Detekt 1.23.8
